op-batcher/batcher/channel_config.go

```diff
@@ -0,0 +1,119 @@
+	//
+	// If 0, duration checks are disabled.
+	MaxChannelDuration uint64
+	// The batcher tx submission safety margin (in #L1-blocks) to subtract from
+	// a channel's timeout and sequencing window, to guarantee safe inclusion of
+	// a channel on L1.
+	SubSafetyMargin uint64
+	// The maximum byte-size a frame can have.
+	MaxFrameSize uint64
+
+	// Target number of frames to create per channel.
+	// For blob transactions, this controls the number of blobs to target adding
+	// to each blob tx.
+	TargetNumFrames int
+
+	// CompressorConfig contains the configuration for creating new compressors.
+	// It should not be set directly, but via the Init*Compressor methods after
+	// creating the ChannelConfig to guarantee a consistent configuration.
+	CompressorConfig compressor.Config
+
+	// BatchType indicates whether the channel uses SingularBatch or SpanBatch.
+	BatchType uint
+
+	// Whether to put all frames of a channel inside a single tx.
+	// Should only be used for blob transactions.
+	MultiFrameTxs bool
+}
+
+// InitCompressorConfig (re)initializes the channel configuration's compressor
+// configuration using the given values. The TargetOutputSize will be set to a
+// value consistent with cc.TargetNumFrames and cc.MaxFrameSize.
+// comprKind can be the empty string, in which case the default compressor will
+// be used.
+func (cc *ChannelConfig) InitCompressorConfig(approxComprRatio float64, comprKind string) {
+	cc.CompressorConfig = compressor.Config{
+		// Compressor output size needs to account for frame encoding overhead
+		TargetOutputSize: MaxDataSize(cc.TargetNumFrames, cc.MaxFrameSize),
+		ApproxComprRatio: approxComprRatio,
+		Kind:             comprKind,
+	}
+}
+
+func (cc *ChannelConfig) InitRatioCompressor(approxComprRatio float64) {
+	cc.InitCompressorConfig(approxComprRatio, compressor.RatioKind)
+}
+
+func (cc *ChannelConfig) InitShadowCompressor() {
+	cc.InitCompressorConfig(0, compressor.ShadowKind)
+}
+
+func (cc *ChannelConfig) InitNoneCompressor() {
+	cc.InitCompressorConfig(0, compressor.NoneKind)
+}
+
+func (cc *ChannelConfig) MaxFramesPerTx() int {
+	if !cc.MultiFrameTxs {
+		return 1
+	}
+	return cc.TargetNumFrames
+}
+
+// Check validates the [ChannelConfig] parameters.
+func (cc *ChannelConfig) Check() error {
+	// The [ChannelTimeout] must be larger than the [SubSafetyMargin].
+	// Otherwise, new blocks would always be considered timed out.
+	if cc.ChannelTimeout < cc.SubSafetyMargin {
+		return ErrInvalidChannelTimeout
+	}
+
+	// The max frame size must at least be able to accommodate the constant
+	// frame overhead.
+	if cc.MaxFrameSize < derive.FrameV0OverHeadSize {
+		return fmt.Errorf("max frame size %d is less than the minimum %d",
+			cc.MaxFrameSize, derive.FrameV0OverHeadSize)
+	}
+
+	if cc.BatchType > derive.SpanBatchType {
+		return fmt.Errorf("unrecognized batch type: %d", cc.BatchType)
+	}
+
+	if nf := cc.TargetNumFrames; nf < 1 {
+		return fmt.Errorf("invalid number of frames %d", nf)
```

Comment:
they could, but since we only compare the error types in tests, but not in actual production code error handling, it's better to not create a bunch of error types that only clutter the package's api.